content/client-lib-development-guide/features.textile

```diff
@@ -1168,6 +1171,8 @@ h4. ClientOptions
 *** @(TO3l4)@ @httpRequestTimeout@ integer - default 10,000 (10s). Timeout for any single HTTP request and response
 *** @(TO3l5)@ @httpMaxRetryCount@ integer - default 3. Max number of fallback hosts to use as a fallback when an HTTP request to the primary host is unreachable or indicates that it is unserviceable
 *** @(TO3l6)@ @httpMaxRetryDuration@ integer - default 15,000 (15s). Max elapsed time in which fallback host retries for HTTP requests will be attempted i.e. if the first default host attempt takes 7s, and then the subsequent fallback retry attempt takes 10s, no further fallback host attempts will be made as the total elapsed time of 17s exceeds the default 15s limit
+*** @(TO3l7)@ @maxMessageSize@ integer - default 65536 (64kB). The maximum size of messages that can be published in one go. That is, the size of the @ProtocolMessage.messages@ or @ProtocolMessage.presence@ array for a realtime publish or presence action, or the message or array of messages being published for a REST publish. The size is defined as the sum over all messages being published of the message @name@, @data@, @clientId@, and @extras@. The size of the @data@ should be measured after being stringified if an @Object@ or @Array@, but before the entire frame is serialized to JSON or msgpack (so e.g. if a binary payload using the JSON encoding, before it is encoded as base64 per @RSL4d1@). For realtime publishes, the default will be overridden by the @maxMessageSize@ in the @connectionDetails@, see "CD2c":#CD2c
+*** @(TO3l8)@ @maxFrameSize@ integer - default 524288 (512kB). The maximum size of a single POST body or WebSocket frame. This is mostly only relevant for `Rest#request` (e.g. for batch publishes), since publishes will hit the @maxMessageSize@ limit before this
```

Comment:
I don't recall agreeing we would have a max frame size at 500kb?  @paddybyers are we going to support batch publishes of up to 500kb then by default?  Doesn't this just provide a way for people then to work around a sensible max size limit we want per request?

Reply:
@mattheworiordan Realtime can limit endpoints to whatever we like. Maybe batch publish 256kB, push publish 16kB, whatever. Having a client lib limit for unknown request types with `Rest#request` doesn't mean the realtime has to allow up to the same value for any and all experimental endpoints we introduce, it's just a backstop to stop the client from sending something several megabytes big to an experimental endpoint, using up our bandwidth for something we know we're going to reject.

So 512kB was picked as a limit that we're very unlikely to want to introduce an experimental endpoint that allows requests larger than that.

I was originally just going to remove the `maxFrameSize` entirely, but then figured it was probably a good idea to have \*something\* to stop the client lib sending multi-megabyte bodies to experimental endpoints (while still giving us flexibility to introduce an endpoint that takes requests larger than the single-message-publish-limit should we want to). Makes sense?

Reply:
@SimonWoolf Ok, fair enough.